### PR TITLE
[ Milestone 7 ] Form

### PR DESCRIPTION
**Pull request by CakeInTech**
- [x] Created a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- [x] Created forms to perform the following functions:
- [x] Created a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- [x] Created a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Users can make a new post.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -3,6 +3,19 @@ def index
...
 
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(author: current_user, title: params[:title], text: params[:text])
+    if @post.save
+      redirect_to start_path
+    else
```
Comment (codepantha) [inline app/controllers/posts_controller.rb]: - [x] [OPTIONAL] Please consider using strong parameters. This is a security practice to help prevent accidentally allowing users to update sensitive model attributes 👍 
Please visit this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) if you need a refresher 😎

❗ Please do the same for the comments controller

File: app/views/comments/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<h>Comment Here</h
+<%= form_with model: @comments, url: comment_path do |form| %>
+  <%= form.text_area :text %>
+  <%= form.submit "Post Comment" %>
```
Comment (codepantha) [inline app/views/comments/new.html.erb:4]: - [x] Clicking the button to post a comment does not work. Please try to fix this bug 🐛 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @MohamedCK 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added forms 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (CakeInTech) [reply app/controllers/posts_controller.rb]: Thank you for the recommendation, I made use of the `permit`. 😄

File: app/views/comments/new.html.erb (hunk 2 above)
Comment (CakeInTech) [reply app/views/comments/new.html.erb:4]: Yes I checked and it seems there was a bug on my controller. Thank you for letting me know. 😄

**Review by uwadonat — APPROVED** [review]
## Hi @MohamedCK 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Good pull request message ✔️
- Good commit history ✔️
209362895-9762f2f8-13cf-4ce0-90b6-a672c4f04196

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**